src/azure-cli-core/azure/cli/core/_identity.py

```diff
@@ -0,0 +1,268 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+import json
+
+from ._environment import get_config_dir
+from .util import get_file_json
+
+from knack.util import CLIError
+from knack.log import get_logger
+
+from azure.identity import (
+    AuthProfile,
+    InteractiveBrowserCredential,
+    DeviceCodeCredential,
+    UsernamePasswordCredential,
+    ClientSecretCredential,
+    CertificateCredential,
+    ManagedIdentityCredential
+)
+
+_CLIENT_ID = '04b07795-8ddb-461a-bbee-02f9e1bf7b46'
+logger = get_logger(__name__)
+
+_SERVICE_PRINCIPAL_ID = 'servicePrincipalId'
+_SERVICE_PRINCIPAL_TENANT = 'servicePrincipalTenant'
+_ACCESS_TOKEN = 'accessToken'
+_SERVICE_PRINCIPAL_CERT_FILE = 'certificateFile'
+_SERVICE_PRINCIPAL_CERT_THUMBPRINT = 'thumbprint'
+
+
+def _load_tokens_from_file(file_path):
+    if os.path.isfile(file_path):
+        try:
+            return get_file_json(file_path, throw_on_empty=False) or []
+        except (CLIError, ValueError) as ex:
+            raise CLIError("Failed to load token files. If you have a repro, please log an issue at "
+                           "https://github.com/Azure/azure-cli/issues. At the same time, you can clean "
+                           "up by running 'az account clear' and then 'az login'. (Inner Error: {})".format(ex))
+    return []
+
+
+def _delete_file(file_path):
+    try:
+        os.remove(file_path)
+    except FileNotFoundError:
+        pass
+
+
+class Identity:
+    """Class to interact with Azure Identity.
+    """
+    def __init__(self, authority, tenant_id):
+        self.authority = authority
+        self.tenant_id = tenant_id
+
+    def login_with_interactive_browser(self):
+        # Use InteractiveBrowserCredential
+        if self.tenant_id:
+            credential, auth_profile = InteractiveBrowserCredential.authenticate(
+                client_id=_CLIENT_ID,
+                tenant_id=self.tenant_id
+            )
+        else:
+            credential, auth_profile = InteractiveBrowserCredential.authenticate(
+                client_id=_CLIENT_ID,
+                silent_auth_only=True
```

Comment:
> silent_auth_only=True [](start = 15, length = 22)

silent_auth_only=True, this should not be related with if tenant_id is none or not.

Reply:
Fixed.